sunpy/map/maputils.py

```diff
@@ -1,7 +1,252 @@
 """
-This submodule provides utility functions to act on `sunpy.map.Map` instances.
+This submodule provides utility functions to act on `sunpy.map.GenericMap` instances.
 """
+from itertools import chain, product
 
+import numpy as np
 
-__all__ = []
+import astropy.units as u
+from astropy.coordinates import SkyCoord
 
+from sunpy.coordinates import Helioprojective
+
+__all__ = ['all_pixel_indices_from_map', 'all_coordinates_from_map',
+           'map_edges', 'contains_full_disk',
+           'is_all_off_disk', 'is_all_on_disk', 'contains_limb',
+           'coordinate_is_on_disk', 'on_disk_bounding_coordinates']
+
+
+def all_pixel_indices_from_map(smap):
+    """
+    Returns pixel pair indices of every pixel in a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    out : `~numpy.array`
+        A numpy array with the all the pixel indices built from the
+        dimensions of the map.
+    """
+    return np.meshgrid(*[np.arange(v.value) for v in smap.dimensions]) * u.pix
+
+
+def all_coordinates_from_map(smap):
+    """
+    Returns the co-ordinates of every pixel in a map.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.GenericMap`
+        A SunPy map.
+
+    Returns
+    -------
+    out : `~astropy.coordinates.SkyCoord`
+        An array of sky coordinates in the coordinate system "coordinate_system".
+    """
+    x, y = all_pixel_indices_from_map(smap)
+    return smap.pixel_to_world(x, y)
+
+
+def map_edges(smap):
+    """
+    Returns the pixel locations of the edges of a rectangular input map.
```

Comment:
```suggestion
    Returns the pixel locations of the edges of a `~sunpy.map.Map` with rectangular data.
```

Reply:
I am not sure I prefer this suggested wording, I am a little confused as to what's being said here. Why does it matter if it's rectangular? Aren't all maps rectangular?

Reply:
I think they are all rectangular.... maybe just drop rectangular?